From 68ac67ea32c58e16acfb8438cca5e30fe4106036 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <alessandro.astone@hotmail.com>
Date: Sun, 2 Sep 2018 02:07:53 +0200
Subject: [PATCH 7/7] libbt: Make sure that we don't load pre-patch when
 looking for patch

 * hw_config_findpatch logic allowed for pre-patch files
   to be loaded in place of patch files.
   Check that the filename is not the pre-patch name.

Change-Id: I19cdf10af21d7d741dcbea42e5a4966d94349d95
---
 src/hardware.c | 43 ++++++++++++++++++++++++-------------------
 1 file changed, 24 insertions(+), 19 deletions(-)

diff --git a/src/hardware.c b/src/hardware.c
index 1e748be..ca978a1 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -558,30 +558,35 @@ static uint8_t hw_config_findpatch(char *p_chip_id_str)
                           FW_PATCHFILE_EXTENSION_LEN) \
                      ) == 0))
                 {
-                    ALOGI("Found patchfile: %s/%s", \
-                        fw_patchfile_path, dp->d_name);
 
-                    /* Make sure length does not exceed maximum */
-                    if ((filenamelen + strlen(fw_patchfile_path)) > \
-                         FW_PATCHFILE_PATH_MAXLEN)
+                    /* Make sure it's not a prepatch */
+                    if(strcmp(fw_prepatch_name, dp->d_name))
                     {
-                        ALOGE("Invalid patchfile name (too long)");
-                    }
-                    else
-                    {
-                        memset(p_chip_id_str, 0, FW_PATCHFILE_PATH_MAXLEN);
-                        /* Found patchfile. Store location and name */
-                        strcpy(p_chip_id_str, fw_patchfile_path);
-                        if (fw_patchfile_path[ \
-                            strlen(fw_patchfile_path)- 1 \
-                            ] != '/')
+                        ALOGI("Found patchfile: %s/%s", \
+                            fw_patchfile_path, dp->d_name);
+
+                        /* Make sure length does not exceed maximum */
+                        if ((filenamelen + strlen(fw_patchfile_path)) > \
+                             FW_PATCHFILE_PATH_MAXLEN)
                         {
-                            strcat(p_chip_id_str, "/");
+                            ALOGE("Invalid patchfile name (too long)");
                         }
-                        strcat(p_chip_id_str, dp->d_name);
-                        retval = TRUE;
+                        else
+                        {
+                            memset(p_chip_id_str, 0, FW_PATCHFILE_PATH_MAXLEN);
+                            /* Found patchfile. Store location and name */
+                            strcpy(p_chip_id_str, fw_patchfile_path);
+                            if (fw_patchfile_path[ \
+                                strlen(fw_patchfile_path)- 1 \
+                                ] != '/')
+                            {
+                                strcat(p_chip_id_str, "/");
+                            }
+                            strcat(p_chip_id_str, dp->d_name);
+                            retval = TRUE;
+                        }
+                        break;
                     }
-                    break;
                 }
             }
         }
-- 
2.17.1

